ktx2Decoder/src/Transcoders/mscTranscoder.ts

```diff
@@ -0,0 +1,114 @@
+import { Nullable } from '../types';
+import { Transcoder, sourceTextureFormat, transcodeTarget } from '../transcoder';
+import { KTX2FileReader, IKTX2_ImageDesc } from '../ktx2FileReader';
+import { WASMMemoryManager } from '../wasmMemoryManager';
+
+declare var MSC_TRANSCODER: any;
+
+declare function importScripts(...urls: string[]): void;
+
+/**
+ * @hidden
+ */
+export class MSCTranscoder extends Transcoder {
+    /**
+     * URL to use when loading the MSC transcoder
+     */
+    public static JSModuleURL = "https://preview.babylonjs.com/ktx2Transcoders/msc_basis_transcoder.js";
+    /**
+     * URL to use when loading the wasm module for the transcoder
+     */
+    public static WasmModuleURL = "https://preview.babylonjs.com/ktx2Transcoders/msc_basis_transcoder.wasm";
+
+    public static UseFromWorkerThread = true;
+
+    private _mscBasisTranscoderPromise: Promise<any>;
+    private _mscBasisModule: any;
+
+    private _getMSCBasisTranscoder() {
+        if (this._mscBasisTranscoderPromise) {
+            return this._mscBasisTranscoderPromise;
+        }
+
+        this._mscBasisTranscoderPromise = new Promise((resolve, reject) => {
+            if (MSCTranscoder.UseFromWorkerThread) {
+                importScripts(MSCTranscoder.JSModuleURL);
+            }
+            WASMMemoryManager.LoadWASM(MSCTranscoder.WasmModuleURL)
+                .then((wasmBinary) => {
+                    MSC_TRANSCODER({ wasmBinary }).then((basisModule: any) => {
+                        basisModule.initTranscoders();
+                        this._mscBasisModule = basisModule;
+                        resolve();
+                    });
+                })
+                .catch((reason) => {
+                    reject(reason);
+                });
+        });
+
+        return this._mscBasisTranscoderPromise;
+    }
+
+    public static CanTranscode(src: sourceTextureFormat, dst: transcodeTarget): boolean {
+        return true;
+    }
+
+    public transcode(src: sourceTextureFormat, dst: transcodeTarget, level: number, width: number, height: number, uncompressedByteLength: number, ktx2Reader: KTX2FileReader, imageDesc: Nullable<IKTX2_ImageDesc>, encodedData: Uint8Array): Promise<Nullable<Uint8Array>> {
+        const isVideo = false;
+
+        return this._getMSCBasisTranscoder().then(() => {
+            const basisModule = this._mscBasisModule;
+
+            const TranscodeTarget: any = basisModule.TranscodeTarget;
+            const TextureFormat: any = basisModule.TextureFormat;
+            const ImageInfo: any = basisModule.ImageInfo;

```

Comment:
I understand this but those 3 variables are really types we get from the module...

It seems clearer to me to have types with a leading upper letter:
```
new ImageInfo(...)
```
instead of
```
new imageInfo(...)
```
Anyway, those variables are used a single time in the code below, so I have simply removed them and replaced them by their values.

Reply:
oh, sorry, I missed that these are types. Looks okay then.